### PR TITLE
[#2443] refactor(web): Restructure web and configure production mode for the static export

### PR DESCRIPTION
**Pull request by LauraXia123**
### What changes were proposed in this pull request?

Reorganising the web folder hierarchy
- Placing application code under the src directory
- Reorganising the component file structure based on layout
- Remove the ui routing folder of the application
- Redefine the main project portal：${application}/index.html
- Configuring `distDir:dist/ui` and `basePath:/ui `parameters for build static files

### Why are the changes needed?

- Distinguish between code files and configuration files
image

- Corresponding file code one by one according to the nest layout
image
image

- Distinguish between production and development modes by configuring a secondary catalogue` /ui`
image

Fix: #2443

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

run the e2e test
image

### CODE REVIEW

**Comment by xunliu** [comment]
@LauraXia123 Please fix Github Action issue.

**Comment by LauraXia123** [comment]
Additional code optimisation content：
Optimise module references, change the way which layer-by-layer downward parameter props to reduce coupling
eg:
image

File: web/src/app/layout.js [hunk 1]
```diff
@@ -3,30 +3,27 @@
  * This software is licensed under the Apache License version 2.
  */
 
-import '@/lib/styles/globals.css'
+import '@/src/lib/styles/globals.css'
```
Comment (ch3yne) [inline web/src/app/layout.js:6]: There's no need to modify all similar lines.
You can set the `compilerOptions` property `"@/*": ["./src/*"]` in the `tsconfig.json`.

File: web/tailwind.config.js
Comment (ch3yne) [inline web/tailwind.config.js:1]: Items line of property`content` need change to `/src`.

File: web/src/app/layout.js [hunk 2]
```diff
@@ -3,30 +3,27 @@
...
+import Loading from './rootLayout/Loading'
+import Layout from './rootLayout/Layout'
 import StyledToast from '../components/StyledToast'
 
 export const metadata = {
   title: 'Gravitino',
   description: 'A high-performance, geo-distributed and federated metadata lake.',
   icons: {
-    icon: '/icons/gravitino.svg'
+    icon: '/ui/icons/gravitino.svg'
```
Comment (ch3yne) [inline web/src/app/layout.js:22]: You can differentiate paths based on the operating environment.

File: web/next.config.js [hunk 3]
```diff
@@ -43,9 +39,10 @@ const nextConfig = {
         }
       }),
   output: process.env.OUTPUT_MODE || 'standalone',
-  distDir: 'dist',
+  basePath: process.env.BASE_PATH,
+  distDir: process.env.DIST_DIR,
   trailingSlash: false,
-  reactStrictMode: false
+  reactStrictMode: true
```
[the remark this answers is not in the document]
Comment (LauraXia123) [reply web/next.config.js:45]: React's [Strict Mode](https://react.dev/reference/react/StrictMode) is a development mode only feature for highlighting potential problems in an application. It helps to identify unsafe lifecycles, legacy API usage, and a number of other features.

Any problems before to set false?

[the remark this answers is not in the document]
Comment (LauraXia123) [reply web/next.config.js:45]: We can find the root cause for why the components or lifecycles will render twice later, and no impact on production patterns, keep the 'true' value for reactStrictMode, is that ok?

**Review by ch3yne — APPROVED** [review]
LGTM, Thanks @LauraXia123.

**Pull request merged**